app/src/component/channelSelector/view.js

```diff
@@ -0,0 +1,243 @@
+import React from 'react';
+import { CLAIM_VALUES, isNameValid } from 'lbry-redux';
+import { ActivityIndicator, Picker, Text, TextInput, TouchableOpacity, View } from 'react-native';
+import Button from 'component/button';
+import Colors from 'styles/colors';
+import Constants from 'constants';
+import Icon from 'react-native-vector-icons/FontAwesome5';
+import Link from 'component/link';
+import channelSelectorStyle from 'styles/channelSelector';
+
+export default class ChannelSelector extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      currentSelectedValue: Constants.ITEM_ANONYMOUS,
+      newChannelName: '',
+      newChannelBid: 0.1,
+      addingChannel: false,
+      creatingChannel: false,
+      newChannelNameError: '',
+      newChannelBidError: '',
+      createChannelError: undefined,
+      showCreateChannel: false,
+    };
+  }
+
+  componentDidMount() {
+    const { channels, fetchChannelListMine, fetchingChannels } = this.props;
+    if (!channels.length && !fetchingChannels) {
+      fetchChannelListMine();
+    }
+  }
+
+  handleCreateCancel = () => {
+    this.setState({ showCreateChannel: false, newChannelName: '', newChannelBid: 0.1 });
+  };
+
+  handlePickerValueChange = (itemValue, itemIndex) => {
+    if (Constants.ITEM_CREATE_A_CHANNEL === itemValue) {
+      this.setState({ showCreateChannel: true });
+    } else {
+      this.handleCreateCancel();
+      this.handleChannelChange(Constants.ITEM_ANONYMOUS === itemValue ? CLAIM_VALUES.CHANNEL_ANONYMOUS : itemValue);
+    }
+    this.setState({ currentSelectedValue: itemValue });
+  };
+
+  handleChannelChange = value => {
+    const { onChannelChange } = this.props;
+    const { newChannelBid } = this.state;
+    const channel = value;
+
+    if (channel === CLAIM_VALUES.CHANNEL_NEW) {
+      this.setState({ addingChannel: true });
+      if (onChannelChange) {
+        onChannelChange(channel);
+      }
+      this.handleNewChannelBidChange(newChannelBid);
+    } else {
+      this.setState({ addingChannel: false });
+      if (onChannelChange) {
+        onChannelChange(channel);
+      }
+    }
+  };
+
+  handleNewChannelNameChange = value => {
+    const { notify } = this.props;
+
+    let newChannelName = value;
+
+    if (newChannelName.startsWith('@')) {
+      newChannelName = newChannelName.slice(1);
+    }
+
+    let newChannelNameError;
+    if (newChannelName.length > 1 && !isNameValid(newChannelName.substr(1), false)) {
+      notify({ message: 'LBRY channel names must contain only letters, numbers and dashes.' });
+    }
+
+    this.setState({
+      newChannelNameError,
+      newChannelName,
+    });
+  };
+
+  handleNewChannelBidChange = newChannelBid => {
+    const { balance, notify } = this.props;
+    let newChannelBidError;
+    if (newChannelBid === 0) {
+      newChannelBidError = __('Your deposit cannot be 0');
```

Comment:
"Please enter a deposit above 0"

(also, if the input is not disallowing negative numbers, the above check should be `< 0` rather than `=== 0`)